core/domain/stats_domain.py

```diff
@@ -16,213 +16,250 @@
 
-        if not isinstance(self.answers_list, list):
+    # NOTE TO DEVELOPERS: do not use the rule_spec_str and answer_str
+    # parameters, as they are here as part of the answer migration.
+    def __init__(self, normalized_answer, interaction_id, answer_group_index,
+                 rule_spec_index, classification_categorization, params,
+                 session_id, time_spent_in_sec, rule_spec_str=None,
+                 answer_str=None):
+        self.normalized_answer = normalized_answer
+        self.interaction_id = interaction_id
+        self.answer_group_index = answer_group_index
+        self.rule_spec_index = rule_spec_index
+        self.classification_categorization = classification_categorization
+        self.params = params
+        self.session_id = session_id
+        self.time_spent_in_sec = time_spent_in_sec
+        self.rule_spec_str = rule_spec_str
+        self.answer_str = answer_str
+
+    @property
+    def json_size(self):
+        """Returns the size of the dict returned by to_dict(). This may not be
+        exactly the size of the returned dict, since it assumes the property in
+        the dict for the json_size is sys.maxint.
+        """
+        return sys.getsizeof(json.dumps(self.to_dict(json_size=sys.maxint)))
+
+    def to_dict(self, json_size=None):
+        submitted_answer_dict = {
+            'answer': self.normalized_answer,
+            'interaction_id': self.interaction_id,
+            'answer_group_index': self.answer_group_index,
+            'rule_spec_index': self.rule_spec_index,
+            'classification_categorization': self.classification_categorization,
+            'params': self.params,
+            'session_id': self.session_id,
+            'time_spent_in_sec': self.time_spent_in_sec,
+            'json_size': self.json_size if not json_size else json_size,
+        }
+        if self.rule_spec_str is not None:
+            submitted_answer_dict['rule_spec_str'] = self.rule_spec_str
+        if self.answer_str is not None:
+            submitted_answer_dict['answer_str'] = self.answer_str
+        return submitted_answer_dict
+
+    @classmethod
+    def from_dict(cls, submitted_answer_dict):
+        return cls(
+            submitted_answer_dict['answer'],
+            submitted_answer_dict['interaction_id'],
+            submitted_answer_dict['answer_group_index'],
+            submitted_answer_dict['rule_spec_index'],
+            submitted_answer_dict['classification_categorization'],
+            submitted_answer_dict['params'],
+            submitted_answer_dict['session_id'],
+            submitted_answer_dict['time_spent_in_sec'],
+            rule_spec_str=submitted_answer_dict.get('rule_spec_str'),
+            answer_str=submitted_answer_dict.get('answer_str'))
+
+    def validate(self):
+        """Validates this submitted answer object."""
+        # TODO(bhenning): Validate the normalized answer against future answer
```

Comment:
#956 generally discusses introducing a new answer type system, possibly with its own validation functions and object orientation. This validate isn't complete without verifying the answer itself is well-formed and that can't be done easily without refactoring objects into answer type classes.